### PR TITLE
Cli upgrade

### PR DESCRIPTION
**Pull request by samanthajayasinghe**

### CODE REVIEW

File: cmd/ocm-backplane/upgrade/cmd.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
+package upgrade
```
Comment (MitaliBhalla) [inline cmd/ocm-backplane/upgrade/cmd.go]: Rename this file to upgrade.go , to maintain the current structure of bp-cli

File: cmd/ocm-backplane/upgrade/cmd.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	"github.com/openshift/backplane-cli/internal/github"
+	"github.com/openshift/backplane-cli/internal/upgrade"
+	"github.com/openshift/backplane-cli/pkg/info"
+	"github.com/spf13/cobra"
+)
+
+func long() string {
+	return strings.Join([]string{
+		"Upgrades the current backplane-cli version based on",
+		"the latest release published in GitLab available for",
```
Comment (MitaliBhalla) [inline cmd/ocm-backplane/upgrade/cmd.go]: Release will not be available on Gitlab anymore, we can just make it something like :
`Latest version release`

File: cmd/ocm-backplane/upgrade/cmd.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+		"your machine's OS and architecture.",
+	}, " ")
+}
+
+var UpgradeCmd = &cobra.Command{
+	Use:   "upgrade",
+	Short: "Upgrades the current backplane-cli version",
```
Comment (MitaliBhalla) [inline cmd/ocm-backplane/upgrade/cmd.go]:
```suggestion
	Short: "Upgrades the backplane-cli to latest version.",
```

File: internal/github/github_test.go [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+	for name, tc := range map[string]struct {
+		Actual          upgrade.Release
+		ExpectedVersion string
+	}{
+		"no releases versions": {
+			Actual:          upgrade.Release{TagName: ""},
+			ExpectedVersion: "",
+		},
+		"single releases versions": {
```
Comment (MitaliBhalla) [inline internal/github/github_test.go]: Perhaps it will be single release version?

File: internal/github/github_test.go [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+			Actual:          upgrade.Release{TagName: "v0.0.1"},
+			ExpectedVersion: "v0.0.1",
+		},
+		"multiple releases versions": {
+			Actual:          upgrade.Release{TagName: "v0.0.2"},
+			ExpectedVersion: "v0.0.2",
+		},
```
Comment (MitaliBhalla) [inline internal/github/github_test.go:36]: What about the cases when API is unresponsive?

**Comment by MitaliBhalla** [comment]
It will also be good to add more comments.

File: cmd/ocm-backplane/upgrade/cmd.go (hunk 1 above)
Comment (samanthajayasinghe) [reply cmd/ocm-backplane/upgrade/cmd.go]: done

File: cmd/ocm-backplane/upgrade/cmd.go (hunk 2 above)
Comment (samanthajayasinghe) [reply cmd/ocm-backplane/upgrade/cmd.go]: done

File: cmd/ocm-backplane/upgrade/cmd.go (hunk 3 above)
Comment (samanthajayasinghe) [reply cmd/ocm-backplane/upgrade/cmd.go]: done

File: internal/github/github_test.go [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+			match:         true,
+			ExpectedAsert: "https://github.com/openshift/backplane-cli/releases/download/v0.0.1/backplane-cli_0.0.1_Darwin_arm64.tar.gz",
+		},
+		"check matching asert for Linux": {
+			LatestRelease: upgrade.Release{
+				TagName: "v0.0.1",
+				Assets: []upgrade.ReleaseAsset{
+					{
+						Name:        "backplane-cli_0.0.1_Linux_arm64.tar.gz",
+						DownloadUrl: "https://github.com/openshift/backplane-cli/releases/download/v0.0.1/backplane-cli_0.0.1_Linux_arm64.tar.gz",
```
Comment (MitaliBhalla) [inline internal/github/github_test.go:108]: This seems hardcoded to me? We can't go ahead with this, we need to add it in a way that it works everytime.
You can refer to something like this https://gitlab.cee.redhat.com/service/backplane-cli/-/blob/master/pkg/info/info.go#L11

File: internal/github/github_test.go [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+					},
+					{
+						Name:        "backplane-cli_0.0.1_Darwin_arm64.tar.gz",
+						DownloadUrl: "https://github.com/openshift/backplane-cli/releases/download/v0.0.1/backplane-cli_0.0.1_Darwin_arm64.tar.gz",
```
Comment (MitaliBhalla) [inline internal/github/github_test.go:129]: This seems hardcoded to me? We can't go ahead with this, we need to add it in a way that it works everytime.
You can refer to something like this https://gitlab.cee.redhat.com/service/backplane-cli/-/blob/master/pkg/info/info.go#L11

File: internal/github/github_test.go [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+			},
+			ExpectedVersion: "",
+			ExpectedAssets:  []upgrade.ReleaseAsset{},
+		},
+		"latest releases versions": {
+			Actual: upgrade.Release{
+				TagName: "v0.0.1",
+				Assets:  []upgrade.ReleaseAsset{},
+			},
+			ExpectedVersion: "v0.0.1",
```
Comment (MitaliBhalla) [inline internal/github/github_test.go:34]: The version here says expected 0.0.1 whereas at the info.go file it is 0.0.0, maybe a hardcoding thing again?

**Comment by MitaliBhalla** [comment]
Also, please comments in the code wherever you can.

**Comment by MitaliBhalla** [comment]
Please add a good description to this PR as well.

**Pull request merged**